### PR TITLE
TW-1457: Migrate android predictive back

### PR DESCRIPTION
**Pull request by nqhhdev**
### Ticket:

- #1457 

### DoD:

- https://docs.flutter.dev/release/breaking-changes/android-predictive-back

### Resolved:

https://github.com/linagora/twake-on-matrix/assets/99852347/036d05a8-1767-4386-a969-097aa9a55af6

### CODE REVIEW

File: lib/widgets/layouts/adaptive_layout/app_adaptive_scaffold_body.dart [hunk 1]
```diff
@@ -25,7 +27,8 @@ class AppAdaptiveScaffoldBody extends StatefulWidget {
       AppAdaptiveScaffoldBodyController();
 }
 
-class AppAdaptiveScaffoldBodyController extends State<AppAdaptiveScaffoldBody> {
+class AppAdaptiveScaffoldBodyController extends State<AppAdaptiveScaffoldBody>
+    with WidgetsBindingObserver {
```
Comment (sherlockvn) [inline lib/widgets/layouts/adaptive_layout/app_adaptive_scaffold_body.dart:31]: why need to use this mixin

Comment (nqhhdev) [reply lib/widgets/layouts/adaptive_layout/app_adaptive_scaffold_body.dart]: Not need, remove it

File: lib/widgets/layouts/adaptive_layout/app_adaptive_scaffold_body.dart [hunk 2]
```diff
@@ -11,6 +12,7 @@ typedef OnOpenSearchPage = Function();
 typedef OnCloseSearchPage = Function();
 typedef OnClientSelectedSetting = Function(Object object, BuildContext context);
 typedef OnDestinationSelected = Function(int index);
+typedef OnPopInvoked = Function(bool);
```
Comment (sherlockvn) [inline lib/widgets/layouts/adaptive_layout/app_adaptive_scaffold_body.dart:15]: please add the return type also. if not, Dart will make return type as dynamic by default.
As the best practice say: https://dart.dev/tools/linter-rules/always_specify_types

Comment (nqhhdev) [reply lib/widgets/layouts/adaptive_layout/app_adaptive_scaffold_body.dart]: done

**Review by sherlockvn — APPROVED** [review]

**Review by hieutbui — APPROVED** [review]

**Pull request merged**